packages/syft/src/syft/service/project/project.py

```diff
@@ -65,34 +76,44 @@ def metadata_to_node_identity() -> List[Callable]:
     return [keep(["id", "verify_key"])]
 
 
-# @serializable()
-# class EventTypes(Enum):
-#     REQUEST = "REQUEST"
-#     USER_CODE = "USER_CODE"
-
-
 class ProjectEvent(SyftObject):
     __canonical_name__ = "ProjectEvent"
     __version__ = SYFT_OBJECT_VERSION_1
 
     id: UID = UID()
-    project_id: UID
     timestamp: DateTime
-    creator_verify_key: SyftVerifyKey
-    prev_event_hash: int
+    project_id: Optional[UID]
+    creator_verify_key: Optional[SyftVerifyKey]
+    prev_event_uid: Optional[UID]
+    prev_signed_event_hash: Optional[int]
+    event_hash: Optional[int]
+    signature: Optional[bytes]  # dont use in signature
 
     @pydantic.root_validator(pre=True)
     def make_timestamp(cls, values: Dict[str, Any]) -> Dict[str, Any]:
         if "timestamp" not in values or values["timestamp"] is None:
             values["timestamp"] = DateTime.now()
         return values
 
-    event_hash: Optional[int]
-    signature: Optional[bytes]
-
     def __hash__(self) -> int:
         return type(self).calculate_hash(self, self.__hash_keys__)
 
+    def rebase(self, project: UID, prev_event: Optional[ProjectEvent]) -> Self:
```

Comment:
Q: I think when we do rebase and the clients are out of sync, I think the entity has to
- recompute the hashes
- modify the parent id
- resign the messages.

Something like how git does, it has also has to be done on the client side, as they possess the signing key

Reply:
I also exploring, on how we could resolve conflicts without resigning